### PR TITLE
python wrapper for s3d

### PR DESCRIPTION
**Pull request by zhiyzuo**
pr for python wrapper plus some utility scripts for s3d.

- structure of the code base: three folders
  - `data/`: store datasets ready for input;
  - `splitted_data/`: to run cross validation with s3d, datasets can be split by using `split_data.py`. the data split will be exported to `splitted_data/{data_name}`;  
  - `s3d`: this folder contains all c++ source code as well as python wrapper scripts, including a quick start notebook (`quickstart.ipynb`) and intermediate as well as result files produced by s3d.

- 🌟 ___python wrapper scripts___
  - `s3d/pys3d.py` is the most important file that wraps both `train` (`PYS3D.fit` method) and `predict_expectation` (`PYS3D.predict` method). `PYS3D. cross_val` and `PYS3D. cross_val_multicore` are the two important functions to facilitate the process of cross validation, using data files (`train.csv`) in `splitted_data`. They takes in a list of `lambda` and max_features (usually set to the number of features) and produces a performance files that records various performance metrics on the ___validation___ set. Another useful function `PYS3D.evaluate` will read in these performance files to evaluate s3d models using the best hyperparameters on the test set (i.e., `test.csv` in `split_data/`), given a specified metric (via arg `cv_metric`.) functions that start with underscores are _inner functions_ used by those directly called by users.
  - `s3d/utils.py` contains a couple of helper function to calculate prediction metrics (both classification and regression) and most importantly, code to visualize s3d model (mostly based on your code in the python notebooks.)

- some other files:
  - `s3d/makefile`: modified so that compilation works in linux
  - `s3d/init` and `s3d/cleanup` are to make folder/file structure setup easier. more details in `quickstart.ipynb`
  - `s3d/requirements.txt`: a list of python packages to guarantee everything will work in the jupyter notebook.

### CODE REVIEW

**Comment by peterfennell** [comment]
Looks good zhiya. I think we can refactor more in the future and even have the user pass a dataframe (which we then save as a .csv and use our functionality) but all that is future work.

Great job!
